src/coreclr/jit/assertionprop.cpp

```diff
@@ -2562,6 +2562,14 @@ GenTree* Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTree* tree)
     ValueNumPair vnPair = tree->gtVNPair;
     ValueNum     vnCns  = vnStore->VNConservativeNormalValue(vnPair);
 
+    // Replace KEEPALIVE(ObjHandle) with NO-OP as the object handle is always
+    // kept alive by the runtime.
```

Comment:
Can this be done in the general walk over nodes (`optAssertionProp`) instead of here?

Nominally this is really a morph transformation, but we cannot use VNs easily in morph.

Reply:
@SingleAccretion looks like `optAssertionProp` is never fired for my example because it has zero assertions? so sounds like that path misses opportunities

Reply:
I've added a separate path for morph

Reply:
Ah, interesting idea, I have more ideas for such generic prop. @SingleAccretion can you take a look at the impl I've just pushed